pkg/helpers/tokencmd/basicauth.go

```diff
@@ -87,6 +89,10 @@ func (c *BasicChallengeHandler) HandleChallenge(requestURL string, headers http.
 		} else {
 			fmt.Fprintf(w, "Authentication required for %s\n", c.Host)
 		}
+		consoleURL, err := guessConsoleURL(c.Host)
+		if err == nil {
+			fmt.Fprintf(w, "Default console URL: %s\n", consoleURL)
```

Comment:
Maybe say it's possibly at that URL, because we allow configuring that through https://github.com/openshift/api/blob/0a6cab75f657a12860a97faaf37ec71cd92530fa/kubecontrolplane/v1/types.go#L36

Reply:
I am not sure if that is a good choice, ie `$API_URL/console` . Even though it can be set I have not found this documented anywhere. I am worried that for most of the users this will not work.

My solution is not great either and would prefer not to merge it. Because we are guessing the apps subdomain and the domain  of the cluster is configurable through ingress. So this will not work in many cases as well.

I think the best consistent solution would be for apiserver to detect a console URL by default if not specified , so the user would always get redirected to the right URL.

Thoughts?

Reply:
ping @soltysh

Reply:
@atiratree maybe we should have a hardcoded `$API_URL/console` which will redirect to appropriate console URL as configured. This would allow us to use that, but still would leave users with a functioning mechanism for exposing the URL where they wish. wdyt?